--- a/drivers/net/wireless/marvell/mwifiex/pcie_quirks.c
+++ b/drivers/net/wireless/marvell/mwifiex/pcie_quirks.c
@@ -20,7 +20,9 @@ static const struct dmi_system_id mwifie
 		.matches = {
 			/* match for SKU here due to generic product name "Surface Pro" */
 			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
+#if LINUX_VERSION_IS_GEQ(4,18,0)
 			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "Surface_Pro_1796"),
+#endif
 		},
 		.driver_data = (void *)QUIRK_FW_RST_D3COLD,
 	},
@@ -29,7 +31,9 @@ static const struct dmi_system_id mwifie
 		.matches = {
 			/* match for SKU here due to generic product name "Surface Pro" */
 			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
+#if LINUX_VERSION_IS_GEQ(4,18,0)
 			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "Surface_Pro_1807"),
+#endif
 		},
 		.driver_data = (void *)QUIRK_FW_RST_D3COLD,
 	},
