From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Varun Gupta <varun.gupta@intel.com>
Date: Tue, 26 May 2026 11:41:55 +0530
Subject: drm/pagemap: Provide compat for dev_pagemap_ops folio_free

Signed-off-by: Varun Gupta <varun.gupta@intel.com>
---
 drivers/gpu/drm/drm_pagemap.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/gpu/drm/drm_pagemap.c b/drivers/gpu/drm/drm_pagemap.c
index b101788..53f8fcf 100644
--- a/drivers/gpu/drm/drm_pagemap.c
+++ b/drivers/gpu/drm/drm_pagemap.c
@@ -1146,10 +1146,17 @@ err_out:
  * This function is a callback used to put the GPU SVM zone device data
  * associated with a page when it is being released.
  */
+#ifdef BPM_DEV_PAGEMAP_OPS_FOLIO_FREE_NOT_PRESENT
+static void drm_pagemap_folio_free(struct page *page)
+{
+	drm_pagemap_zdd_put(page->zone_device_data);
+}
+#else
 static void drm_pagemap_folio_free(struct folio *folio)
 {
 	drm_pagemap_zdd_put(folio->page.zone_device_data);
 }
+#endif
 
 /**
  * drm_pagemap_migrate_to_ram() - Migrate a virtual range to RAM (page fault handler)
@@ -1175,7 +1182,11 @@ static vm_fault_t drm_pagemap_migrate_to_ram(struct vm_fault *vmf)
 }
 
 static const struct dev_pagemap_ops drm_pagemap_pagemap_ops = {
+#ifdef BPM_DEV_PAGEMAP_OPS_FOLIO_FREE_NOT_PRESENT
+	.page_free = drm_pagemap_folio_free,
+#else
 	.folio_free = drm_pagemap_folio_free,
+#endif
 	.migrate_to_ram = drm_pagemap_migrate_to_ram,
 };
 
-- 
2.43.0

