--- a/drivers/net/wireless/atmel/at76c50x-usb.c
+++ b/drivers/net/wireless/atmel/at76c50x-usb.c
@@ -513,11 +513,19 @@ exit:
 
 /* LED trigger */
 static int tx_activity;
+#if LINUX_VERSION_IS_LESS(4,15,0)
+static void at76_ledtrig_tx_timerfunc(unsigned long data);
+#else
 static void at76_ledtrig_tx_timerfunc(struct timer_list *unused);
+#endif
 static DEFINE_TIMER(ledtrig_tx_timer, at76_ledtrig_tx_timerfunc);
 DEFINE_LED_TRIGGER(ledtrig_tx);
 
+#if LINUX_VERSION_IS_LESS(4,15,0)
+static void at76_ledtrig_tx_timerfunc(unsigned long data)
+#else
 static void at76_ledtrig_tx_timerfunc(struct timer_list *unused)
+#endif
 {
 	static int tx_lastactivity;
 
