From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Mon, 3 Feb 2025 16:52:34 +0530
Subject: autotools/m4: manage FOP_UNSIGNED_OFFSET availability

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/gpu/drm/xe/xe_device.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/xe/xe_device.c b/drivers/gpu/drm/xe/xe_device.c
index dfc062d..afe2cbf 100644
--- a/drivers/gpu/drm/xe/xe_device.c
+++ b/drivers/gpu/drm/xe/xe_device.c
@@ -259,7 +259,9 @@ static const struct file_operations xe_driver_fops = {
 #ifdef CONFIG_PROC_FS
 	.show_fdinfo = drm_show_fdinfo,
 #endif
+#ifndef BPM_FOP_UNSIGNED_OFFSET_NOT_PRESENT
 	.fop_flags = FOP_UNSIGNED_OFFSET,
+#endif
 };
 
 #if !defined(CPTCFG_OOT_BACKPORT)
-- 
2.43.0

