--- a/drivers/net/wireless/mediatek/mt7601u/dma.c
+++ b/drivers/net/wireless/mediatek/mt7601u/dma.c
@@ -75,7 +75,11 @@ bad_frame:
 
 static void mt7601u_rx_process_seg(struct mt7601u_dev *dev, u8 *data,
 				   u32 seg_len, struct page *p,
+#if LINUX_VERSION_IS_GEQ(4,19,0)
 				   struct list_head *list)
+#else
+				   struct sk_buff_head *list)
+#endif
 {
 	struct sk_buff *skb;
 	struct mt7601u_rxwi *rxwi;
@@ -136,8 +140,14 @@ mt7601u_rx_process_entry(struct mt7601u_
 	u32 seg_len, data_len = e->urb->actual_length;
 	u8 *data = page_address(e->p);
 	struct page *new_p = NULL;
-	LIST_HEAD(list);
 	int cnt = 0;
+#if LINUX_VERSION_IS_GEQ(4,19,0)
+	LIST_HEAD(list);
+#else
+	struct sk_buff_head list;
+
+	__skb_queue_head_init(&list);
+#endif
 
 	if (!test_bit(MT7601U_STATE_INITIALIZED, &dev->state))
 		return;
