--- a/drivers/net/wireless/mac80211_hwsim.c
+++ b/drivers/net/wireless/mac80211_hwsim.c
@@ -5318,7 +5318,11 @@ done:
 }
 
 /* Generic Netlink operations array */
+#if LINUX_VERSION_IS_GEQ(5,10,0)
 static const struct genl_small_ops hwsim_ops[] = {
+#else
+static const struct genl_ops hwsim_ops[] = {
+#endif
 	{
 		.cmd = HWSIM_CMD_REGISTER,
 		.validate = GENL_DONT_VALIDATE_STRICT | GENL_DONT_VALIDATE_DUMP,
@@ -5362,8 +5366,13 @@ static struct genl_family hwsim_genl_fam
 	.policy = hwsim_genl_policy,
 	.netnsok = true,
 	.module = THIS_MODULE,
+#if LINUX_VERSION_IS_GEQ(5,10,0)
 	.small_ops = hwsim_ops,
 	.n_small_ops = ARRAY_SIZE(hwsim_ops),
+#else
+	.ops = hwsim_ops,
+	.n_ops = ARRAY_SIZE(hwsim_ops),
+#endif
 	.resv_start_op = HWSIM_CMD_DEL_MAC_ADDR + 1,
 	.mcgrps = hwsim_mcgrps,
 	.n_mcgrps = ARRAY_SIZE(hwsim_mcgrps),
