--- a/drivers/net/wireless/intersil/orinoco/orinoco.h
+++ b/drivers/net/wireless/intersil/orinoco/orinoco.h
@@ -208,6 +208,7 @@ int orinoco_stop(struct net_device *dev)
 void orinoco_set_multicast_list(struct net_device *dev);
 int orinoco_change_mtu(struct net_device *dev, int new_mtu);
 void orinoco_tx_timeout(struct net_device *dev, unsigned int txqueue);
+void bp_orinoco_tx_timeout(struct net_device *dev);
 
 /********************************************************************/
 /* Locking and synchronization functions                            */
