--- a/drivers/net/wireless/marvell/libertas/if_spi.c
+++ b/drivers/net/wireless/marvell/libertas/if_spi.c
@@ -235,9 +235,14 @@ static int spu_read(struct if_spi_card *
 		spi_message_add_tail(&dummy_trans, &m);
 	} else {
 		/* Busy-wait while the SPU fills the FIFO */
+#if LINUX_VERSION_IS_GEQ(5,5,0)
 		reg_trans.delay.value =
 			DIV_ROUND_UP((100 + (delay * 10)), 1000);
 		reg_trans.delay.unit = SPI_DELAY_UNIT_USECS;
+#else
+		reg_trans.delay_usecs =
+			DIV_ROUND_UP((100 + (delay * 10)), 1000);
+#endif /* >= 5.5 */
 	}
 
 	/* read in data */
