--- a/drivers/net/wireless/microchip/wilc1000/spi.c
+++ b/drivers/net/wireless/microchip/wilc1000/spi.c
@@ -284,10 +284,14 @@ static int wilc_spi_tx(struct wilc *wilc
 		struct spi_transfer tr = {
 			.tx_buf = b,
 			.len = len,
+#if LINUX_VERSION_IS_GEQ(5,5,0)
 			.delay = {
 				.value = 0,
 				.unit = SPI_DELAY_UNIT_USECS
 			},
+#else
+			.delay_usecs = 0,
+#endif /* >= 5.5 */
 		};
 		char *r_buffer = kzalloc(len, GFP_KERNEL);
 
@@ -327,10 +331,14 @@ static int wilc_spi_rx(struct wilc *wilc
 		struct spi_transfer tr = {
 			.rx_buf = rb,
 			.len = rlen,
+#if LINUX_VERSION_IS_GEQ(5,5,0)
 			.delay = {
 				.value = 0,
 				.unit = SPI_DELAY_UNIT_USECS
 			},
+#else
+			.delay_usecs = 0,
+#endif /* >= 5.5 */
 
 		};
 		char *t_buffer = kzalloc(rlen, GFP_KERNEL);
@@ -371,10 +379,14 @@ static int wilc_spi_tx_rx(struct wilc *w
 			.tx_buf = wb,
 			.len = rlen,
 			.bits_per_word = 8,
+#if LINUX_VERSION_IS_GEQ(5,5,0)
 			.delay = {
 				.value = 0,
 				.unit = SPI_DELAY_UNIT_USECS
 			},
+#else
+			.delay_usecs = 0,
+#endif /* >= 5.5 */
 
 		};
 
